app/src/lib/components/CommitDragItem.svelte

```diff
@@ -1,131 +1,60 @@
-	const project = getContext(Project);
-	const branch = maybeGetContextStore(Branch);
-
-	function acceptAmend(commit: Commit | RemoteCommit) {
-		if (!$branch) return;
-
-		if (commit instanceof RemoteCommit) {
-			return () => false;
-		}
-		return (data: any) => {
-			if (!project.ok_with_force_push && commit.isRemote) {
-				return false;
-			}
-
-			if (commit.isIntegrated) {
-				return false;
-			}
-
-			if (data instanceof DraggableHunk && data.branchId === $branch.id) {
-				return true;
-			} else if (data instanceof DraggableFile && data.branchId === $branch.id) {
-				return true;
-			} else {
-				return false;
-			}
-		};
-	}
-
-	function onAmend(commit: Commit | RemoteCommit) {
-		if (!$branch) return;
-
-		return (data: any) => {
-			if (data instanceof DraggableHunk) {
-				const newOwnership = `${data.hunk.filePath}:${data.hunk.id}`;
-				branchController.amendBranch($branch.id, commit.id, newOwnership);
-			} else if (data instanceof DraggableFile) {
-				if (data.file instanceof LocalFile) {
-					// this is an uncommitted file change being amended to a previous commit
-					const newOwnership = filesToOwnership(data.files);
-					branchController.amendBranch($branch.id, commit.id, newOwnership);
-				} else if (data.file instanceof RemoteFile) {
-					// this is a file from a commit, rather than an uncommitted file
-					const newOwnership = filesToSimpleOwnership(data.files);
-					if (data.commit) {
-						branchController.moveCommitFile($branch.id, data.commit.id, commit.id, newOwnership);
-					}
-				}
-			}
-		};
+	interface Props {
+		commit: Commit | RemoteCommit;
+		children: Snippet;
 	}
 
-	function acceptSquash(commit: Commit | RemoteCommit) {
-		if (!$branch) return;
+	const { commit, children }: Props = $props();
 
-		if (commit instanceof RemoteCommit) {
-			return () => false;
-		}
-		return (data: any) => {
-			if (!(data instanceof DraggableCommit)) return false;
-			if (data.branchId !== $branch.id) return false;
-
-			if (data.commit.isParentOf(commit)) {
-				if (data.commit.isIntegrated) return false;
-				if (data.commit.isRemote && !project.ok_with_force_push) return false;
-				return true;
-			} else if (commit.isParentOf(data.commit)) {
-				if (commit.isIntegrated) return false;
-				if (commit.isRemote && !project.ok_with_force_push) return false;
-				return true;
-			} else {
-				return false;
-			}
-		};
-	}
+	const branch = maybeGetContextStore(Branch);
 
-	function onSquash(commit: Commit | RemoteCommit) {
+	const actions = $derived.by(() => {
 		if (!$branch) return;
 
-		if (commit instanceof RemoteCommit) {
-			return () => false;
-		}
-		return (data: DraggableCommit) => {
-			if (data.commit.isParentOf(commit)) {
-				branchController.squashBranchCommit(data.branchId, commit.id);
-			} else if (commit.isParentOf(data.commit)) {
-				branchController.squashBranchCommit(data.branchId, data.commit.id);
-			}
-		};
-	}
+		return commitDragActionsFactory.build($branch, commit);
+	});
```

Comment:
I was considering saying something like `$derived($branch && commitDragActionsFactory.build($branch, commit))`, but I decided that I really just want to keep everything as simple as possible, and being explicit and having that sort of guard clause to me just expressed the intent much better

Reply:
Yup, makes it more readable too, etc. etc. Good call :+1: